docs/source/cli.rst

```diff
@@ -23,3 +54,159 @@ Following are the detail for the shell :term:`CLI` which provides the same featu
+------------------------
+CLI and Client Examples
+------------------------
+
+Following sections present different typical usage of the :ref:`cli_commands` and :ref:`client_commands`.
+Operations are equivalent between the :term:`CLI` and Python client.
+
+Note that more operations and option parameters are available,
+and are not all necessarily represented in below examples.
+
+For each of the following examples, the client is created as follows:
+
+.. code-block:: python
+
+    client = WeaverClient(url="{WEAVER_URL}")
+
+
+.. _cli_example_deploy:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Deploy Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. todo:: example
+
+.. _cli_undeploy:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Undeploy Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. todo:: example
+
+
+.. _cli_example_getcap:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+GetCapabilities Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Accomplishes the :ref:`GetCapabilities <proc_op_getcap>` request to obtain a list of available :term:`Process`.
+
+.. code-block:: shell
+
+    weaver capabilities -u {WEAVER_URL}
+
+.. code-block:: python
+
+    client.capabilities()
+
+Sample Output:
+
+.. literalinclude:: ../../weaver/wps_restapi/examples/local_process_listing.json
+    :language: json
+
+
+.. _cli_example_describe:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+DescribeProcess Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Accomplishes the :ref:`DescribeProcess <proc_op_describe>` request to obtain the :term:`Process` definition.
+
+.. code-block:: shell
+
+    weaver describe -u {WEAVER_URL} -p jsonarray2netcdf
+
+.. code-block:: python
+
+    client.describe("jsonarray2netcdf")
+
+Sample Output:
+
+.. literalinclude:: ../../weaver/wps_restapi/examples/local_process_description_ogc_api.json
+    :language: json
+
+.. _cli_example_execute:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Execute Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Accomplishes the :ref:`Execute <proc_op_execute>` request to obtain launch a :term:`Job`
+with the specified :term:`Process` and provided inputs.
+
+.. todo:: example
+
+.. _cli_example_dismiss:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Dismiss Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. todo:: example
+
+
+.. _cli_example_status:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+GetStatus Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+.. todo:: example
+
+.. _cli_example_monitor:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Monitor Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. todo:: example
+
+
+.. _cli_example_results:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Results Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+
+.. todo:: example
+
+.. _cli_example_upload:
+
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Upload Example
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+This operation allows manual upload of a local file to the :term:`Vault`.
+
+.. note::
+    When running the :ref:`execute` operation, any detected local file reference will be automatically uploaded
```

Comment:
So Weaver doesn't rely anymore on extern hosting?

Reply:
This is specific only when using the CLI. Local files that are not accessible for the remote weaver are uploaded to it. 
If an URL is provided, it will act as usual.